druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/RocksDBExtractionNamespaceCacheFactory.java

```diff
@@ -69,70 +75,33 @@ public String call() {
         };
     }
 
+    public void tryResetRunnerOrLog(RocksDBExtractionNamespace extractionNamespace) {
+        try {
+            if (!extractionNamespace.cacheActionRunner.isEmpty() && Objects.isNull(cacheActionRunner)) {
+                cacheActionRunner = CacheActionRunner.class.cast(
+                        Class.forName(extractionNamespace.cacheActionRunner).newInstance());
+                LOG.debug("Populated a new CacheActionRunner with description " + cacheActionRunner.toString());
+            } else if (Objects.isNull(cacheActionRunner)) {
```

Comment:
We need an 'else' statement here, otherwise we risk NPE if neither condition is met.